app/views/handlers/question.py

```diff
@@ -113,8 +113,40 @@ def get_next_location_url(self):
             self._return_to_block_id,
         )
 
+    # pylint: disable=protected-access
     def _get_answers_for_question(self, question_json) -> dict[str, Any]:
         answer_ids = self._schema.get_answer_ids_for_question(question_json)
+        if question_json.get("dynamic_answers_list_id"):
```

Comment:
This should check the dynamic answers key, also the change below shouldn't repeat for normal answers, it should be:

```py
if dynamic:
	handle_dynamic()

handle_static_if_any()
```